src/main.rs

```diff
@@ -85,15 +112,28 @@ fn run_with_file_map(conf: Config, log: Logger) -> Result<(), Error> {
             file_map.insert(fname, file);
         }
     }
+    if conf.multithreaded {
+        run_multithreaded(conf, log, move |msg| {
+            let shark = msg.shark.replace(domain_prefix.clone().as_str(), "");
+            let shard = msg.shard;
+            println!("shark: {}, shard: {}", shark, shard);
 
-    sharkspotter::run(conf, log, |moray_obj, shark, shard| {
-        let shark = shark.replace(domain_prefix.clone().as_str(), "");
-        println!("shark: {}, shard: {}", shark, shard);
+            let file =
+                file_map.get_mut(&filename(shark.as_str(), shard)).unwrap();
```

Comment:
nit: Comment about why this unwrap is safe?

Reply:
fixed.